File: 250110/일렬로 서있는 소 2/cattle-in-a-rowing-up-2.cpp
#include <iostream>
using namespace std;

int main() {
    // Please write your code here.
    int N, arr[100];
    cin >> N;
    for (int i=0; i<N; i++) {
        cin >> arr[i];
    }
    int cnt = 0;
    for (int i=0; i<N; i++) {
        for (int j=i+1; j<N; j++) {
            for (int k=j+1; k<N; k++) {
                if (arr[i] <= arr[j] && arr[j] <= arr[k]) cnt++;
            }
        }
    }
    cout << cnt;
    return 0;
}

File: 250628/일렬로 서있는 소 2/cattle-in-a-rowing-up-2.py
N = int(input())
arr = list(map(int, input().split()))

cnt = 0
for i in range(N):
    for j in range(i+1, N):
        for k in range(j+1, N):
            if not (arr[i] <= arr[j] and arr[j] <= arr[k]):
                continue
            cnt += 1

print(cnt)